cfbs/cfbs_config.py

```diff
@@ -314,3 +316,82 @@ def add_command(
             msg = msg[4:]  # Remove the '\n - ' part of the message
         changes_made = count > 0
         return Result(0, changes_made, msg)
+
+    def input_command(self, module_name, input_data):
+        def _check_keys(keys, input_data):
+            for key in keys:
+                if key not in input_data:
+                    user_error(
+                        "Expected attribute '%s' in input definition: %s"
+                        % (key, pretty(input_data))
+                    )
+
+        def _input_string(input_data):
+            _check_keys(["question"], input_data)
+            response = prompt_user(
+                self.non_interactive,
+                input_data["question"],
+                default=input_data.get("default"),
+            )
+            return response
+
+        def _input_list(input_data):
+            _check_keys(["subtype", "while"], input_data)
+            subtype = input_data["subtype"]
+
+            if isinstance(subtype, list):
+                result = []
+
+                def _input_elements():
+                    result = OrderedDict()
+                    for element in subtype:
+                        _check_keys(["type", "label", "question", "key"], element)
+                        if element["type"] != "string":
+                            user_error(
+                                "Subtype of type '%s' not supported for type list"
+                                % element["type"]
+                            )
+                        result[element["key"]] = _input_string(element)
+                    return result
```

Comment:
This nested function somewhere inside another function is weird. I'd move it above the `_input_list()` function with one argument for the `subtype`.